protocol/message_persistence.go

```diff
@@ -2381,42 +2381,61 @@ func (db sqlitePersistence) GetDeletes(messageID string, from string) ([]*Delete
 	if err != nil {
 		return nil, err
 	}
+	defer rows.Close()
 
 	var messages []*DeleteMessage
-
 	for rows.Next() {
 		d := &DeleteMessage{}
 		err := rows.Scan(&d.Clock, &d.ChatId, &d.MessageId, &d.From, &d.ID)
 		if err != nil {
 			return nil, err
 		}
 		messages = append(messages, d)
-
 	}
 	return messages, nil
 }
 
-func (db sqlitePersistence) SaveDeleteForMe(deleteForMeMessage DeleteForMeMessage) error {
-	_, err := db.db.Exec(`INSERT INTO user_messages_deleted_for_mes (clock, message_id, source, id) VALUES(?,?,?,?)`, deleteForMeMessage.Clock, deleteForMeMessage.MessageId, deleteForMeMessage.From, deleteForMeMessage.ID)
+func (db sqlitePersistence) SaveOrUpdateDeleteForMeMessage(deleteForMeMessage *protobuf.DeleteForMeMessage) error {
+	_, err := db.db.Exec(`INSERT OR REPLACE INTO user_messages_deleted_for_mes (clock, message_id) 
+    SELECT ?,? WHERE NOT EXISTS (SELECT 1 FROM user_messages_deleted_for_mes WHERE message_id = ? AND clock >= ?)`,
```

Comment:
@qfrank have you checked this query? Looking into it I am not sure that it works.

Reply:
> have you checked this query

yes